### ISSUE TITLE
[ENH] add ReduceOnPlateau callbacks for the learning rate to all deep learning classifiers

### ISSUE THREAD

**Issue by hadifawaz1999** [aeon-toolkit/aeon#320]
as a default callback it should be added to not overfit the models, when getting in InceptionTime before there was an issue with this callbacks during test checks. To see if it works now.

- [x] InceptionTime
- [x] FCN
- [x] ResNet
- [x] MLP

The other classifiers do not use them by default, for more info check [dl4tsc](https://github.com/hfawaz/dl-4-tsc/tree/master/classifiers)
